File: Randoms/875. Koko Eating Bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l=1
        r=max(piles)
        result=r

        while l<=r:
            k= (l+r)//2
            hours = 0
            for x in piles:
                hours += math.ceil(x/k)
            
            if hours<=h:
                result = min(r,k)
                r = k-1
            else:
                l = k+1
        return result

File: Randoms/875. Koko Eating Bananas.ts
function minEatingSpeed(piles: number[], h: number): number {
    let l=1;
    let r= Math.max(...piles);
    let result = r;

    while (l<=r){
        let k=Math.floor((l+r)/2);
        let hours = 0;

        for (let x of piles){
            hours += Math.ceil(x/k);
        } 

        if (hours<=h){
            result = Math.min(r,k)
            r = k-1
        }

        else{
            l = k+1
        }
    }
    
    return result;
};
